src/static/js/collab_client.js

```diff
@@ -182,6 +163,26 @@ const getCollabClient = (ace2editor, serverVars, initialUserInfo, options, _pad)
         });
   };
 
+  const serverMessageTaskQueue = new class {
+    constructor() {
+      this._q = Promise.resolve();
+    }
+
+    async enqueue(fn) {
+      const p = this._q.then(async () => {
+        while (true) {
+          const composing = editor.getInInternationalComposition();
+          if (!composing) break;
+          await composing;
+        }
+        await fn(); // It's OK if fn() doesn't return a Promise.
+      });
+      // Use .catch() to prevent rejections from halting the queue.
+      this._q = p.catch(() => {});
+      return await p;
```

Comment:
Maybe joining the `.catch` to the promise chain would make it easier to understand.
```js
const p = this._q
  .then(/* ... */)
  .catch(() => {});
  
this._q = p;
return await this._q;
```

Reply:
I don't want to do that because it swallows all errors. I want the caller to see a Promise rejection if `fn()` throws. I'll add a comment.

Reply:
Great! Now I see what you did.

Reply:
Done.